### ISSUE TITLE
await for unused result: Expected an assignment or function call and instead saw an expression

### ISSUE THREAD

**Issue by miqmago** [eslint/eslint#7294]
* **ESLint Version:** 2.7.0
* **Node Version:** 4.4.2
* **npm Version:** 3.8.5

**What parser (default, Babel-ESLint, etc.) are you using?**
    "eslint": "^2.7.0",
    "eslint-config-airbnb": "^6.2.0",
    "eslint-config-angular": "^0.5.0",
    "eslint-plugin-angular": "^1.0.1",
    "eslint-plugin-react": "^5.1.1",

**Please show your full configuration:**

*.eslintrc*
```json
{
    "parser": "babel-eslint",
    "extends": "airbnb",
    "rules": {
        "indent": [2, 4, { "SwitchCase": 1, "VariableDeclarator": {"var": 2, "let": 2, "const": 3} }]
    }
}
```
*.babelrc*
```json
{
    "presets": ["es2015", "stage-0"]
}
```

**What did you do? Please include the actual source code causing the issue.**

This happens when the result of a promise function is not needed for next steps:
```js
await resultNotNeeded(bla, foo); // Produces an error: no-unused-expressions
const dummy = await resultNotNeeded(bla, foo); // Also produces an error: no-unused-vars
```
I know the option of `// eslint-disable-line no-unused-expressions` but I think this case is not a syntax error. And the second expression for me it's not an option, eslint is ok here, as `dummy` is unused.

**What did you expect to happen?**
No error on the first case `await resultNotNeeded(bla, foo);` (as second option is not an option).

**What actually happened? Please include the actual, raw output from ESLint.**

Eslint errors:

> [eslint] Expected an assignment or function call and instead saw an expression. (no-unused-expressions)

> [eslint] 'dummy' is defined but never used (no-unused-vars)

**Comment by not-an-aardvark** [comment]
Thanks for the issue.

I think this bug was fixed in https://github.com/eslint/eslint/pull/7175, which was released in ESLint v3.6.0.

Could you upgrade to the latest version of ESLint and try again?

**Comment by miqmago** [comment]
Thanks, the problem here was that npm was outdated, so when trying to install `eslint@latest` it was installing:

```
$ sudo npm i eslint -g
├─┬ eslint@2.13.1 
```

I've updated npm@3.10.7 (3.10.8 is not working) and finally I could install `eslint@3.7.0`.
